### ISSUE TITLE
[Feature Request] option to control width gestimation of tables.

### ISSUE THREAD

**Issue by terefang** [jgm/pandoc#10433]
today `simple_tables` and `pipe_tables` as well as HTML-Tables are subject to an automatic mechanism gestimating their column widths by various means.

there are various usecases that either want gestimation to be performed always or never.

* https://github.com/jgm/pandoc/issues/10427
* https://github.com/jgm/pandoc/issues/10111

for the sake of continuity also the current behaviour should be preserved most likely as the default.

**Describe your proposed improvement and the problem it solves.**

the proposal is to have an option called `--smart-table-widths` that take the following parameters:

* "auto" being the behavior as it is now, should be the default.
* "off" never compute them as for @sphakka case (above).
* "on" always compute them as for my case.(above).

in my "always" case the gestimated widths become available to lua-filters and the typst target to ensure proper layout regardless of table and column sizes.

in @sphakka "never" case the gestimated widths are discarded in favor of the reflow capabilities of the HTML/CSS target.

**Comment by jgm** [comment]
I believe that only `pipe_tables` is subject to this.  Simple tables never have widths calculated; multiline tables and  grid tables always hav them.  So this only affects pipe tables.

**Comment by jgm** [comment]
Accordingly, the option, if there is one, should be `--pipe-table-widths=smart|always|never`.

**Comment by terefang** [comment]
sound reasonable to me.

**Comment by bpj** [comment]
see https://github.com/jgm/pandoc/issues/10427#issuecomment-2521209222

**Comment by Mathnerd314** [comment]
Not sure if this is actually related, but, as mentioned in the user guide there are "ugly" pipe tables, like:
```
fruit|price
---|---
apple|2.05
pear|1.37
orange|3.09
```
As an option I would like to have Pandoc output these "ugly" pipe tables, i.e. skip the width calculation, use three dashes for each column in the table header, and just output the pipe character or maybe pipe plus spaces like `apple | 2.05` for each row.

**Comment by terefang** [comment]
if i see this correctly that would be the `--pipe-table-widths=never` case

**Comment by jgm** [comment]
> if i see this correctly that would be the `--pipe-table-widths=never` case.

No.  The proposal above was for an option that affects *parsing* of tables, controlling whether width information is computed.  The request above has to do with rendering.  It is irrelevant to this issue.

**Comment by terefang** [comment]
ok ... sorry for my miss-interpretation

**Comment by Mathnerd314** [comment]
>  The request above has to do with rendering. It is irrelevant to this issue.

OK I opened #10453.

**Comment by thekief** [comment]
@terefang to understand your proposal, when using `off`, **pandoc** would use `auto` for Typst's table, right?

**Comment by terefang** [comment]
hmmm .....

referencing https://typst.app/docs/reference/model/table/

in that case i would suggest the table option to be "`columns: auto,` " — yes

**Comment by terefang** [comment]
hello @jgm 

so far i have been able to solve my span and alignment issues with a filter -- https://gist.github.com/terefang/44a7a6cd1c7b7bbd6c1d172e8c28c89c

but the width issue needs parser support.

may i propose a non-invasive addition the the colspec ? https://pandoc.org/lua-filters.html#type-colspec

could you add a third component, actually the length of the header separator ?

so for the table:

```
| col 1 | col 2 | col3 |
| :----- | :----: | ---: |
| v1 | v2 | v3 |
```

the colspecs would be

- [ "AlignLeft", 0, 6 ]
- [ "AlignCenter", 0, 6 ]
- [ "AlignRight", 0, 4 ]


 as a hint for filters, the width component could then be rewritten to the needs of the user.

i think this could also be valuable for other table types.

what you think ?

**Comment by terefang** [comment]
after more than a year, i am still encountering problems related to this issue. also in interaction with the "--columns=<N>" option.

any chance this width gestimator might get implemented or have there been other enhancements or changes i am not aware of ?
